lib/edgedb/protocol/codecs/range.ex

```diff
@@ -33,12 +33,17 @@ defimpl EdgeDB.Protocol.Codec, for: EdgeDB.Protocol.Codecs.Range do
     codec = CodecStorage.get(codec_storage, codec)
     flags = encode_range_flags(range)
 
-    data = [
-      Codec.encode(codec, range.lower, codec_storage),
-      Codec.encode(codec, range.upper, codec_storage)
-    ]
-
-    [<<IO.iodata_length(data)::uint32(), flags::uint8()>> | data]
+    data =
+      Enum.reject(
+        [
+          <<flags::uint8()>>,
+          range.lower && Codec.encode(codec, range.lower, codec_storage),
+          range.upper && Codec.encode(codec, range.upper, codec_storage)
+        ],
+        &is_nil/1
+      )
+
+    [<<IO.iodata_length(data)::uint32()>> | data]
```

Comment:
Yeah, this seems to match the logic in the JS driver:

https://github.com/edgedb/edgedb-js/blob/cbea3a24ccdd284524e05bacf01f0252910adab0/packages/driver/src/codecs/range.ts#L48-L65